stubs/D3DShot/d3dshot/dll/d3d.pyi

```diff
@@ -0,0 +1,226 @@
+import sys
+from _typeshed import Incomplete
+from ctypes import Structure, _CArgObject, c_int32, c_uint, c_void_p
+from ctypes.wintypes import FLOAT, UINT
+from typing_extensions import TypeAlias
+
+from d3dshot.dll.dxgi import IDXGIAdapter
+
+# mypy does not support os.name checks, while pyright does https://github.com/python/mypy/issues/13002
+# import os
+# if os.name == "nt":  # noqa: Y002
```

Comment:
Is there an advantage to using `os.name` instead of `sys.platform` here? If not, I think we can safely do without this comment; I'm not sure it really adds much :)

```suggestion
```

Reply:
I'm not sure. It's what `ctypes` does. So I just wanted to mirrored that to stay accurate, then realized mypy doesn't support it.

Reply:
Emulating the runtime as much as possible is Good and Virtuous. But, in this case, I think it's fairly obvious why we're using `sys.platform` instead of mimicking the runtime exactly.

PEP 484 [specifically calls out `sys.platform` checks](https://peps.python.org/pep-0484/#version-and-platform-checking) as something type checkers are supposed to be able to understand. Any other way of checking for platform-specific behaviour is, from the perspective of the `typing` system, undefined behaviour. Pyright has chosen to go above and beyond the spec here in _also_ implementing support for `os.name` checks, which is awesome -- but that's not something that type checkers are _expected_ to necessarily understand.